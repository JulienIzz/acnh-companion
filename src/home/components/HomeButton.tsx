import {StyleSheet, View, TouchableOpacity} from 'react-native';
import React from 'react';

const BUTTON_TOTAL_HEIGHT = 160;
const BUTTON_TOTAL_WIDTH = 120;
const BUTTON_ICON_HEIGHT_AND_WIDTH = 100;

export const HomeButton = () => {
  <View style={styles.container}>
    <TouchableOpacity style={styles.touchable}></TouchableOpacity>
  </View>;
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    height: BUTTON_TOTAL_HEIGHT,
    width: BUTTON_TOTAL_WIDTH,
  },
  touchable: {
    height: BUTTON_ICON_HEIGHT_AND_WIDTH,
    width: BUTTON_ICON_HEIGHT_AND_WIDTH,
  },
});
