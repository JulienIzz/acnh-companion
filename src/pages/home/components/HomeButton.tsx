import {
  StyleSheet,
  View,
  TouchableOpacity,
  Image,
  Dimensions,
  ImageSourcePropType,
} from 'react-native';
import React from 'react';
import {
  MAX_GRID_WIDTH,
  GRID_WIDTH_RATIO,
  NUMBER_OF_COLUMNS,
} from '../HomeScreen';
import {IconTypography} from '../../../theme/typography/Typography';

const BUTTON_TOTAL_HEIGHT = 160;
const BUTTON_TOTAL_WIDTH =
  Dimensions.get('window').width > MAX_GRID_WIDTH
    ? (MAX_GRID_WIDTH * GRID_WIDTH_RATIO) / NUMBER_OF_COLUMNS
    : (GRID_WIDTH_RATIO * Dimensions.get('window').width) / NUMBER_OF_COLUMNS;
const BUTTON_ICON_HEIGHT_AND_WIDTH = 100;

export interface ButtonData {
  navigationTarget: string;
  imgPath: ImageSourcePropType;
  label: string;
  navigation?: any;
}

export const HomeButton = ({
  navigationTarget,
  imgPath,
  label,
  navigation,
}: ButtonData) => {
  return (
    <View style={styles.container}>
      <TouchableOpacity
        style={styles.touchable}
        onPress={() => navigation.navigate(navigationTarget)}>
        <Image style={styles.image} source={imgPath} />
      </TouchableOpacity>
      <IconTypography text={label} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    height: BUTTON_TOTAL_HEIGHT,
    width: BUTTON_TOTAL_WIDTH,
  },
  touchable: {
    height: BUTTON_ICON_HEIGHT_AND_WIDTH,
    width: BUTTON_ICON_HEIGHT_AND_WIDTH,
  },
  image: {
    width: '100%',
    height: '100%',
  },
});
