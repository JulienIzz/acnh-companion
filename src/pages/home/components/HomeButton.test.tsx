import {HomeButton} from './HomeButton';
import React from 'react';
import {render, screen} from '@testing-library/react-native';

jest.mock('@react-navigation/native');

const MOCK_BUTTON_LABEL = 'Bouton numéro 1';
const MOCK_BUTTON_PATH = require('../img/bugs.png');
const MOCK_BUTTON_TARGET = 'Test';

test('icon grid with pressable buttons', async () => {
  render(
    <HomeButton
      label={MOCK_BUTTON_LABEL}
      imagePath={MOCK_BUTTON_PATH}
      navigationTarget={MOCK_BUTTON_TARGET}
    />,
  );

  const ButtonLabel = screen.getByText('Bouton numéro 1');

  expect(ButtonLabel).toBeTruthy();
  expect(screen.toJSON()).toMatchSnapshot();
});
