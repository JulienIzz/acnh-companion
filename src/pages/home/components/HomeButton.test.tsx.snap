// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon grid with pressable buttons 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "height": 160,
      "width": 150,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "height": 100,
        "opacity": 1,
        "width": 100,
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/pages/home/img/bugs.png",
        }
      }
      style={
        {
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </View>
  <Text
    style={
      {
        "color": "black",
        "fontSize": 14,
        "textAlign": "center",
      }
    }
  >
    Bouton num√©ro 1
  </Text>
</View>
`;
